RIOT-OS/RIOT
# treewide: model newlib as a FEATURE

### Contribution description

This PR models `newlib` as a feature and makes use of `FEATURES_REQUIRED_ANY` to allow requesting any of both `picolibc` and `newlibc` while only having one selected.

This way applications can enforce one of them with `FEATURES_REQUIRED += %`. IMO only applications should enforce this, otherwise, a conflict is likely to arise, `FEATURES_CONFLICT` is added to prevent this.

### Testing procedure

Verify that dependencies are still correct. Some here

```
→ BOARD=samr21-xpro make -C examples/gnrc_minimal/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```
```
→ BOARD=hifive1b make -C examples/gnrc_minimal/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```
```
→ RIOT_CI_BUILD=1 BOARD=hifive1b make -C examples/gnrc_minimal/ info-modules | grep lib
newlib
newlib_nano
newlib_syscalls_fe310
```

```
→ BOARD=nucleo-l011k4 make -C examples/hello-world/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```

```
→ BOARD=nucleo-l011k4 make -C examples/hello-world/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```

```
→ BOARD=samd10-xmini make -C examples/hello-world/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```

```
→ BOARD=stm32f030f4-demo make -C examples/hello-world/ info-modules | grep lib
picolibc
picolibc_syscalls_default
```

```
FEATURES_REQUIRED=newlib BOARD=hifive1b make -C examples/hello-world/ info-modules | grep lib
newlib
newlib_nano
newlib_syscalls_fe310
```

```
BOARD=esp32-wroom-32 make -C examples/gnrc_minimal/ info-modules | grep lib
newlib
newlib_nano
newlib_syscalls_default
```

### Issues/PRs references

Split from #15832
